src/applications/Utilities/src/Utilities.cxx

```diff
@@ -405,49 +405,71 @@ int algorithmsRunner()
   }
   else if (requestedAlgorithm == Dicom2Nifti)
   {
-    auto readDicomImage = cbica::ReadImage< TImageType >(inputImageFile);
-    if (!readDicomImage)
-    {
-      std::cout << "Dicom Load Failed" << std::endl;
-      return EXIT_FAILURE;
-    }
-
-    cbica::WriteImage< TImageType>(readDicomImage, outputImageFile);
-
-    if (!targetImageFile.empty())
-    {
-      if (!cbica::ImageSanityCheck< TImageType >(readDicomImage, cbica::ReadImage< TImageType >(targetImageFile)))
-      {
-        std::cerr << "Input image and target image physical space mismatch.\n";
-        return EXIT_FAILURE;
-      }
-      else
-      {
-        auto diffFilter = itk::Testing::ComparisonImageFilter< TImageType, TImageType >::New();
-        auto inputImage = cbica::ReadImage< TImageType >(inputImageFile);
-        auto size = inputImage->GetBufferedRegion().GetSize();
-        diffFilter->SetValidInput(cbica::ReadImage< TImageType >(targetImageFile));
-        diffFilter->SetTestInput(inputImage);
-        diffFilter->VerifyInputInformationOn();
-        diffFilter->SetDifferenceThreshold(testThresh);
-        diffFilter->SetToleranceRadius(testRadius);
-        diffFilter->UpdateLargestPossibleRegion();
-
-        size_t totalSize = 1;
-        for (size_t d = 0; d < TImageType::ImageDimension; d++)
-        {
-          totalSize *= size[d];
-        }
-
-        std::cout << "Total Voxels/Pixels in Image: " << totalSize << "\n";
-        std::cout << "Number of Difference Voxels : " << diffFilter->GetNumberOfPixelsWithDifferences() << "\n";
-        std::cout << "Percentage of Diff Voxels   : " << diffFilter->GetNumberOfPixelsWithDifferences() * 100 / totalSize << "\n";
-        std::cout << "Minimum Intensity Difference: " << diffFilter->GetMinimumDifference() << "\n";
-        std::cout << "Maximum Intensity Difference: " << diffFilter->GetMaximumDifference() << "\n";
-        std::cout << "Average Intensity Difference: " << diffFilter->GetMeanDifference() << "\n";
-        std::cout << "Overall Intensity Difference: " << diffFilter->GetTotalDifference() << "\n";
-      }
-    }
+    //get folder from input dicom image
+	dicomFolderPath = cbica::getFilenamePath(inputImageFile);
+
+	// construct path to dcm2niix for debug/release modes and different OS
+	std::string m_exe; 
+#ifdef CAPTK_PACKAGE_PROJECT
+	#if WIN32
+		m_exe = cbica::getExecutablePath() + "/dcm2niix.exe";
+	#else
+		m_exe = cbica::getExecutablePath() + "/dcm2niix";
+	#endif
```

Comment:
To add here:
```cpp
#ifndef __APPLE__
    m_exe = cbica::getExecutablePath() + "/dcm2niix";
#else
    m_exe = cbica::getExecutablePath() + "../Resources/bin/dcm2niix"
#endif
```

Reply:
Initially I was also thinking that we would need that check, but since the Utilities and dcm2niix are both in the same directory, it works. We don't need the `if __APPLE__` check.

Reply:
Oh okay!